import urllib2  import socket        class MyException(Exception):          pass        try:      restr = urllib2.urlopen("https://data.btcchina.com/data/historydata?limit=10", timeout = 5)      print restrexcept urllib2.URLError, e:      if isinstance(e.reason, socket.timeout):          raise MyException("There was an error: %r" % e)      else:          # reraise the original error          raise